client/state/posts/selectors.js

```diff
@@ -90,11 +97,17 @@ export const getSitePost = createSelector(
  * @return {?Array}         Posts for the post query
  */
 export function getSitePostsForQuery( state, siteId, query ) {
-	if ( ! state.posts.queries[ siteId ] ) {
+	const manager = state.posts.queries[ siteId ];
+	if ( ! manager ) {
+		return null;
+	}
+
+	const posts = manager.getItems( query );
+	if ( ! posts ) {
 		return null;
 	}
 
-	return state.posts.queries[ siteId ].getItems( query );
+	return posts.map( ( post ) => getNormalizedPost( state, post.global_ID ) );
```

Comment:
Hmm, definitely an interesting idea. Higher-order reducers seem like they could be a good fit for this use-case. I think there might be a few too many implementation details (serialization, queries tracked by site, etc) that would make this difficult to implement, but could be worth exploring.

Now that you mention it though, I wonder if I could refactor the approach I introduced in #6040 this way, or at a minimum to reuse existing reducers instead of custom implementations for tracking current user and/or relevant post information.